### PR TITLE
#1184: export warnings and errors from azure as github comment

### LINKED ISSUE #1184: Explore reporting warnings/errors back as comments in GitHub from Azure pipelines
**What Needs to be Done?**

It would be nice if we can get warnings to report back as a comment in GitHub from an Azure pipeline. I can export a file from the container runs with a list of the warnings/errors from the run. We just need a way to get that back to GitHub.

### PR DESCRIPTION
**Pull request by jstrzebonski**
Fixes #1184

Still TODO:
- [x] address @PhilMiller comment about warnings in nvidia builds
- [x] add to comment information which build puts given comment - without that, it's hard to say anything from that warnings and errors
- [x] comment on all the quirks and hacks that had to be done to make it work - there are not a lot of them, but still, they might not be that obvious to everyone
- [x] check if failed builds also correctly comment on PR
- [x] remove `ninja: build stopped: subcommand failed.` from the report
- [x] move changes from `azure-clang-8-alpine-mpich.yml` to `azure-workflow-template.yml` and regenerate all the azure ymls
- [x] handle situation when there aren't any errors or warnings
- [x] trim too long output
- [x] generate errors/warnings file only if `vt_ci_build` is set
- [x] remove `System.Debug: true` from azure ymls
- [x] Nvidia 11 still produces a lot of "deprecated" warnings while 10.1 doesn't
- [x] test if all the azure builds comment properly (gcc, clang, intel, nvidia)

### CODE REVIEW

**Comment by PhilMiller** [comment]
Among other things to adjust as part of this PR, we may need to add `-Wno-deprecated-gpu-targets` for our Nvidia builds, so that we don't get a warning about compiler arguments for every file compiled.

**Comment by jstrzebonski** [comment]
Yes! Almost there!

Still TODO:
- [ ] address @PhilMiller comment about warnings in nvidia builds
- [ ] add to a comment information which build puts given comment - without that it's hard to say anything from that warnings and errors
- [ ] comment on all the quirks and hacks that had to be done to make it work - there are not a lot of them, but still they might not be that obvious to everyone
- [ ] move changes from `azure-clang-8-alpine-mpich.yml` to `azure-workflow-template.yml` and regenerate all the azure ymls
- [ ] what to do with `build_cpp.sh`? Now it might be a bit noise when used locally - probably no one needs additional file with errors and warnings. Should there be some variable that tells if such file should be created?
- [ ] check if failed builds also correctly comment on PR
- [ ] of course in the end test if all the azure builds comment properly (gcc, clang, intel, nvidia)

File: cmake/nvcc_no_deprecated_gpu_targets.cmake [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Uses flag "-Wno-deprecated-gpu-targets" for NVCC builds
```
Comment (jstrzebonski) [inline cmake/nvcc_no_deprecated_gpu_targets.cmake:1]: Maybe `Add flag ...`?

Comment (jstrzebonski) [reply cmake/nvcc_no_deprecated_gpu_targets.cmake]: Done.

**Comment by jstrzebonski** [comment]
asdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadasasdsadas

**Comment by jstrzebonski** [comment]
OK, so after my naive experimentation I concluded, that flag `-Wno-deprecated-gpu-targets` is missing from nvcc_wrapper's possible `cuda_args` values. After adding it here -> https://github.com/kokkos/nvcc_wrapper/blob/master/nvcc_wrapper.in#L187 and of course setting it in CMake the warning seems to be gone.

So two questions:
* I am right? :- ] I don't have much experience with this wrapper, so maybe there are better ways to make this flag work
* If I'm right, how do we want to solve it? By simply creating a PR to the wrapper code? Is it still supported by anyone? Or maybe we should fork it to our organization?

@lifflander @PhilMiller, what do you think?

**Comment by PhilMiller** [comment]
If that wrapper won't pass the flag through, it's definitely worth posting a PR. It's actively maintained and used.

For us, we'll probably want to apply it in the copy of Kokkos used for our CI in anticipation of it being accepted updated.

**Comment by jstrzebonski** [comment]
@PhilMiller 

I forked https://github.com/kokkos/nvcc_wrapper and fix passing -Wno-deprecated-gpu-targets on a branch, but again, description of this repo says it's deprecated:

image



Also in the last PR ( https://github.com/kokkos/nvcc_wrapper/pull/34 ) it is said that this repo is no longer maintained:

image

What we want to do with this? Do we want to use our fork for builds?

**Comment by PhilMiller** [comment]
Ah, ok, we should move our CI builds to using `nvcc_wrapper` from the main `kokkos/kokkos` repository, and post a PR against that if still necessary.

**Comment by jstrzebonski** [comment]
My proposition is to get fixed nvcc_wrapper from our kokkos fork, to get rid of that wall of warnings, and after our patch gets to next kokkos release just update the dockerfile.

**Comment by jstrzebonski** [comment]
OK, I think it's ready to review. Although shouldn't be merged before DARMA-tasking/vt#1209

**Comment by lifflander** [comment]
I am rebuilding the nvcc images

**Comment by cz4rs** [comment]
@jstrzebonski #1209 was merged, most of the warnings should be gone after rebase

**Comment by jstrzebonski** [comment]
> @jstrzebonski #1209 was merged, most of the warnings should be gone after rebase

Yeap, I've just rebased like 15 minutes ago. Thanks! Most of the builds are warnings-free now :- ]

**Review by lifflander — APPROVED** [review]

**Comment by lifflander** [comment]
Approved to merge once all the builds pass. The nvcc images are refreshed with the nvcc_wrapper changes.

**Comment by lifflander** [comment]
Just restarted all the builds with the new images

**Comment by lifflander** [comment]
@jstrzebonski I just realized that this might end up being a lot of comments. Is there a way to update the comment easily once a new push happens and we have a new set of errors/warning? Do you think that would be useful?

**Comment by cz4rs** [comment]
> @jstrzebonski I just realized that this might end up being a lot of comments. Is there a way to update the comment easily once a new push happens and we have a new set of errors/warning? Do you think that would be useful?

- in regards to warnings, having `-Werror` enabled in CI reduces the number of comments significantly (to the point of obsoleting part of this PR when we have warnings turned into errors in all builds)
- on a similar note, maybe we could skip the below message in case of a successful build (again, no warnings allowed with `-Werror` and failed build is clearly visible in the required checks list):
Screenshot from 2021-01-14 14-30-42

**Review by cz4rs — APPROVED** [review]
the code looks good, the desired behavior might need to be discussed further

:+1:  for the small cleanup in `SetCXXCompilerFlags.cmake`

**Comment by jstrzebonski** [comment]
> @jstrzebonski I just realized that this might end up being a lot of comments. Is there a way to update the comment easily once a new push happens and we have a new set of errors/warning? Do you think that would be useful?

I agree with @cz4rs on this. Firstly, there is no easy way to do it - you have to know the comment ID you want to edit, and I don't think there's a way to somehow remember them and map each comment with each build. Secondly - we have `-Werror` for almost all builds now.

**Comment by jstrzebonski** [comment]
> > @jstrzebonski I just realized that this might end up being a lot of comments. Is there a way to update the comment easily once a new push happens and we have a new set of errors/warning? Do you think that would be useful?
> 
>     * in regards to warnings, having `-Werror` enabled in CI reduces the number of comments significantly (to the point of obsoleting part of this PR when we have warnings turned into errors in all builds)
> 
>     * on a similar note, maybe we could skip the below message in case of a successful build (again, no warnings allowed with `-Werror` and failed build is clearly visible in the required checks list):
>       Screenshot from 2021-01-14 14-30-42

I would opt for that. After @cz4rs added `-Werror` if any warning shows up, the build will go red. I think there isn't any more a need for a message saying that the build is warnings-free.

> edit: on a second thought, warnings that are not coming directly from compiler could still show up - currently there is one coming from CMake regularly:
> 
> ```
> CMake Warning:
>   Manually-specified variables were not used by the project:
> 
>     MI_INTERPOSE
>     MI_OVERRIDE
> ```

Currently comments only gather errors and warnings from the compilation step, not configuration. I would have to rethink the whole implementation to support this. I cannot say how hard would that be.

**Comment by jstrzebonski** [comment]
> edit: regarding the failed builds, is that `grep` error? they seemed to pass fine earlier

I don't think so, the grep worked just fine till yesterday.
I'm investigating this.

**Comment by PhilMiller** [comment]
Once we've got `-Werror` enabled broadly, having the build failure message automatically posted to the PR could be helpful. I find the number of clicks to get through to the actual CI transcript a bit annoying. On the flip side, with our C++, even a single error could be a *lot* of text

**Comment by PhilMiller** [comment]
The warnings about unused variables in unity builds suggest that we should probably be passing `-Wno-unused-variable` to those. Or maybe it's just gcc-5 that's overly sensitive to that.

**Comment by jstrzebonski** [comment]
> The warnings about unused variables in unity builds suggest that we should probably be passing `-Wno-unused-variable` to those. Or maybe it's just gcc-5 that's overly sensitive to that.

Those variables are inside `#pragma GCC diagnostic ignored "-Wunused-variable"` block. To bad gcc-5 doesn't respect that :'- (

**Comment by jstrzebonski** [comment]
Well, actually it was the grep problem. I forgot that returns not-zero if nothing's found :- |

**Comment by jstrzebonski** [comment]
And what about those comments?
image

I think me and @cz4rs agree that putting them in PR could be safely skipped now because of `-Werror`.
@lifflander @PhilMiller what do you think?

**Comment by lifflander** [comment]
> And what about those comments?
> image
> 
> I think me and @cz4rs agree that putting them in PR could be safely skipped now because of `-Werror`.
> @lifflander @PhilMiller what do you think?

Yes, I think they can be skipped now with `-Werror`

**Pull request merged**